File: go/02-control-flow/loops.go
// الحلقات في Go - Loops
package main

import "fmt"

func main() {
	// ===== حلقة for أساسية =====
	fmt.Println("=== حلقة for أساسية ===")
	for i := 0; i < 5; i++ {
		fmt.Printf("العدد: %d\n", i)
	}

	// ===== for كـ while =====
	fmt.Println("\n=== for كـ while ===")
	عداد := 0
	for عداد < 5 {
		fmt.Printf("العداد: %d\n", عداد)
		عداد++
	}

	// ===== التكرار على شريحة =====
	fmt.Println("\n=== التكرار على شريحة ===")
	فواكه := []string{"تفاح", "موز", "برتقال"}
	for فهرس, فاكهة := range فواكه {
		fmt.Printf("%d: %s\n", فهرس, فاكهة)
	}

	// ===== التكرار على خريطة =====
	fmt.Println("\n=== التكرار على خريطة ===")
	درجات := map[string]int{
		"أحمد":  85,
		"محمد":  92,
		"علي":   78,
	}
	for اسم, درجة := range درجات {
		fmt.Printf("%s: %d\n", اسم, درجة)
	}

	// ===== break و continue =====
	fmt.Println("\n=== break و continue ===")
	for i := 0; i < 10; i++ {
		if i == 3 {
			continue // تخطي 3
		}
		if i == 7 {
			break // إيقاف عند 7
		}
		fmt.Println(i)
	}

	// ===== حلقة لا نهائية =====
	fmt.Println("\n=== حلقة مع عداد ===")
	مجموع := 0
	for i := 1; i <= 100; i++ {
		مجموع += i
	}
	fmt.Printf("مجموع 1 إلى 100 = %d\n", مجموع)
}

File: python/02-control-flow/loops.py
# الحلقات في بايثون - Loops

# ===== حلقة for =====
print("=== حلقة for ===")
for i in range(5):
    print(f"العدد: {i}")

# التكرار على قائمة
فواكه = ["تفاح", "موز", "برتقال"]
for فاكهة in فواكه:
    print(f"الفاكهة: {فاكهة}")

# التكرار مع الفهرس
for فهرس, فاكهة in enumerate(فواكه):
    print(f"{فهرس}: {فاكهة}")

# ===== حلقة while =====
print("\n=== حلقة while ===")
عداد = 0
while عداد < 5:
    print(f"العداد: {عداد}")
    عداد += 1

# ===== break و continue =====
print("\n=== break و continue ===")
for i in range(10):
    if i == 3:
        continue  # تخطي 3
    if i == 7:
        break     # إيقاف عند 7
    print(i)

# ===== حلقة for مع else =====
print("\n=== for مع else ===")
for i in range(5):
    print(i)
else:
    print("انتهت الحلقة بدون break!")

# ===== list comprehension =====
print("\n=== list comprehension ===")
مربعات = [x**2 for x in range(1, 6)]
print(f"المربعات: {مربعات}")

أعداد_زوجية = [x for x in range(20) if x % 2 == 0]
print(f"الأعداد الزوجية: {أعداد_زوجية}")

# ===== الحلقات المتداخلة =====
print("\n=== الحلقات المتداخلة ===")
for i in range(1, 4):
    for j in range(1, 4):
        print(f"{i} × {j} = {i*j}")
